src/core/sagas/utils.js

```diff
@@ -1,19 +1,17 @@
-/* @flow */
 import { ErrorHandler } from 'core/errorHandler';
-import log from 'core/logger';
-import type { ApiStateType } from 'core/reducers/api';
+import defaultLog from 'core/logger';
 
-export function createErrorHandler(id: string): typeof ErrorHandler {
+
+export function createErrorHandler(id, { log = defaultLog } = {}) {
   return new ErrorHandler({
     id,
     // Make sure the dispatch() method can't be used. A saga will yield
     // put(action) instead so this shouldn't cause a problem.
-    dispatch: () => log.error(
-      'ErrorHandler cannot dispatch from a saga'),
+    dispatch: () => log.error('ErrorHandler cannot dispatch from a saga'),
   });
 }
 
-// Convenience function to extract API info.
-export function getApi(state: {| api: ApiStateType |}): ApiStateType {
-  return state.api;
+// Convenience function to extract state info.
+export function getState(state) {
```

Comment:
I think an annotation of `api: ApiStateType` would be really helpful here, especially since `state` is now completely non-intuitive. You mentioned that it's only half-documented since we don't have an `auth` definition but half is better than nothing. How could someone even look at this function and know what to do with `state`?

Also, I'm fine with not adding `/* @flow */` to new files but removing it from existing ones? You need to provide a solid argument for that and I don't see one here. It's hard enough for us to get decent Flow coverage; what's the point of reducing our coverage? If it's slowing you down on this patch, just let me know and I can fix it up on master once you land it.